### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by jothepro**
Nothing special, just a small incremental update of the documentation:

- add section about comments in idl (#18)
- remove build instructions from `README.md`
- add link to docs in `README.md`
- add `.idea` folder to `.gitignore`
- fix broken link in "code usage" section
- add basic release documentation for maintainers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,61 +35,3 @@ Make sure to [put it on your PATH permanently](https://stackabuse.com/how-to-per
 ### Windows
 
 Download the [latest released batch file](https://github.com/cross-language-cpp/djinni-generator/releases/latest/download/djinni.bat), copy it where you want and [put the folder on your PATH](https://www.architectryan.com/2018/03/17/add-to-the-path-on-windows-10/).
-
-
-## Building from source
```
Comment (jothepro) [inline README.md:37]: This section has been moved to the developer guide in the mkdocs documentation

File: docs/generated-code-usage.md [hunk 2]
```diff
@@ -30,9 +30,8 @@ somewhere in your code:
     System.loadLibrary("YourLibraryName");
     // The name is specified in Android.mk / build.gradle / Makefile, depending on your build system.
 
-If you package your native library in a jar, you can also use `com.dropbox.djinni.NativeLibLoader` 
-to help unpack and load your lib(s).  See the [Localhost README](example/localhost/README.md)
-for details.
+If you package your native library in a jar, you can also use the [`NativeLibLoader`](https://github.com/cross-language-cpp/djinni-support-lib/blob/main/java/com/dropbox/djinni/NativeLibLoader.java)
```
Comment (jothepro) [inline docs/generated-code-usage.md:33]: I opened [an issue about](https://github.com/cross-language-cpp/djinni-support-lib/issues/8) the `NativeLibLoader`. Not sure what this is about.

Comment (a4z) [reply docs/generated-code-usage.md:33]: We have our own lib loader at work, so I also don't know about that one. But it's good to look at it

**Review by a4z — APPROVED** [review]
Thanks!

**Pull request merged**